x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -0,0 +1,259 @@
+            builder.startObject()
+                .startObject(MapperService.SINGLE_MAPPING_NAME)
+                    .startObject("_source")
+                        .field("enabled", false)
+                    .endObject()
+                    .startObject("properties")
+                        .startObject(policy.getEnrichKey())
+                            .field("type", keyType)
+                            .field("index", true)
+                            .field("doc_values", false)
+                        .endObject()
+                    .endObject()
+                .endObject()
+            .endObject();
+
+            return builder;
+        } catch (IOException ioe) {
+            throw new UncheckedIOException("Could not render enrich mapping", ioe);
+        }
+    }
+
+    private void prepareAndCreateEnrichIndex() {
+        long nowTimestamp = nowSupplier.getAsLong();
+        String enrichIndexName = getEnrichIndexBase(policyName) + "-" + nowTimestamp;
+        // TODO: Settings for localizing enrich indices to nodes that are ingest+data only
+        Settings enrichIndexSettings = Settings.EMPTY;
+        CreateIndexRequest createEnrichIndexRequest = new CreateIndexRequest(enrichIndexName, enrichIndexSettings);
+        createEnrichIndexRequest.mapping(MapperService.SINGLE_MAPPING_NAME, resolveEnrichMapping(policy));
+        logger.debug("Policy [{}]: Creating new enrich index [{}]", policyName, enrichIndexName);
+        client.admin().indices().create(createEnrichIndexRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(CreateIndexResponse createIndexResponse) {
+                transferDataToEnrichIndex(enrichIndexName);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private void transferDataToEnrichIndex(final String destinationIndexName) {
+        logger.debug("Policy [{}]: Transferring source data to new enrich index [{}]", policyName, destinationIndexName);
+        // Filter down the source fields to just the ones required by the policy
+        final Set<String> retainFields = new HashSet<>();
+        retainFields.add(policy.getEnrichKey());
+        retainFields.addAll(policy.getEnrichValues());
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(retainFields.toArray(new String[0]), new String[0]);
+        if (policy.getQuery() != null) {
+            searchSourceBuilder.query(QueryBuilders.wrapperQuery(policy.getQuery().getQuery()));
+        } else {
+            searchSourceBuilder.query(QueryBuilders.matchAllQuery());
+        }
+        ReindexRequest reindexRequest = new ReindexRequest()
+            .setDestIndex(destinationIndexName)
+            .setSourceIndices(policy.getIndexPattern());
+        reindexRequest.getSearchRequest().source(searchSourceBuilder);
```

Comment:
The xcontent type in the enrich index should be smile. So we will need to add the following statement here:

```
reindexRequest.getDestination().source(new BytesArray(), XContentType.SMILE)
```

The `destination` field inside `ReindexRequest` is now actually used, but used as a prototype for specific properties. The source isn't used and can therefore be empty in the above statement. (See `TransportReindexAction#buildRequest(...)` method)